delta_backend/src/FHIRParser.py

```diff
@@ -21,18 +41,21 @@ def _scanValuesForMatch(self, parent, matchValue):
             return False
 
     # locate an index for an item in a list
-    def _locateListId(self, parent, locator):
-        fieldList = locator.split(":")
+    def _locateListId(self, parent, locator, expression_type, expression_rule: str = ""):
+        fieldList = locator.split(":", 1)
         nodeId = 0
         index = 0
         try:
             while index < len(parent):
-                for key in parent[index]:
-                    if (parent[index][key] == fieldList[1]) or (key == fieldList[1]):
+                for key, value in parent[index].items():
+                    if (
+                        (value == fieldList[1] or key == fieldList[1])
+                        and self._validate_expression_rule(expression_type, expression_rule, parent[index])
+                    ):
                         nodeId = index
                         break
                     else:
-                        if self._scanValuesForMatch(parent[index][key], fieldList[1]):
+                        if self._scanValuesForMatch(value, fieldList[1]):

```

Comment:
This branch is confusing me a bit. If we don't find a match in the keys / values of the object we're searching, we then start searching through nested values too? So should we be calling `_validate_expression_rule` here?

Kind of irrelevant because I don't think this functionality to search nested values is ever used. The whole thing feels a bit over-engineered and inefficient to me 😅

Reply:
Anyway, it's not new code, so I'm happy to ignore for now. We can look to tidy this whole thing up once we're done with the essential go live work.